src/Model/GServer.php

```diff
@@ -2048,19 +2051,22 @@ public static function setProtocol(int $gsid, int $protocol)
 	 *
 	 * @param int $gsid Server id
 	 * @return int
-	 * @throws Exception
+	 * @throws InvalidArgumentException If $gsid is not valid
+	 * @throws ResultException If $gsid cannot be found in table
 	 */
 	public static function getProtocol(int $gsid): int
 	{
-		if (empty($gsid)) {
-			return null;
+		if ($gsid < 1) {
+			throw new InvalidArgumentException('gsid=' . $gsid . ' is invalid');
```

Comment:
Why are you throwing an error? Doing so will interrupt the message delivery in some cases.

Reply:
`$gsid` (`gserver.id`) can never be lower 1. A SQL query on such record will always end with an empty result. So it is better to avoid these calls and save some SQL statements.

Reply:
Also you are stopping the message delivery.

Reply:
Both calls to `GServer::getProtocol` already check for `$gsid > 0` before calling and they are returning null anyway in the other case. This method should keep returning `null` as an alternative and the outer checks need to be included in this method.